cmake/toolchain/arcmwdt/Kconfig

```diff
@@ -1,6 +1,9 @@
 # Copyright (c) 2022 Synopsys
 # SPDX-License-Identifier: Apache-2.0
 
+config TOOLCHAIN_ARCMWDT
+	def_bool y
```

Comment:
this is actually a lot of boilerplate being copied and new files added just for this.

We should create a `cmake/toolchain/Kconfig` file with content:
```
config TOOLCHAIN_$(ZEPHYR_TOOLCHAIN_VARIANT_UPPER)
	def_bool y
```

and then source it here:
```diff
--- a/Kconfig.zephyr
+++ b/Kconfig.zephyr
@@ -52,6 +52,7 @@ source "drivers/Kconfig"
 source "lib/Kconfig"
 source "subsys/Kconfig"
 
+source "cmake/toolchain/Kconfig"
 osource "$(TOOLCHAIN_KCONFIG_DIR)/Kconfig"
 
 menu "Build and Link Features"
```

remember to create an upper version and export to Kconfig similar to this:
```diff
+  ZEPHYR_TOOLCHAIN_VARIANT_UPPER=${ZEPHYR_TOOLCHAIN_VARIANT_UPPER}
```
https://github.com/zephyrproject-rtos/zephyr/blob/8e18c5ce1d75023e1386c646a0616774d6202d98/cmake/modules/kconfig.cmake#L129

later we can then replace all uses of `if ZEPHYR_TOOLCHAIN_VARIANT = "<name>"` with just `if TOOLCHAIN_<NAME>` then deprecate and remove the old export.

Reply:
The issue with having:

```
config TOOLCHAIN_$(ZEPHYR_TOOLCHAIN_VARIANT_UPPER)
	def_bool y

```

Is that it doesn't allow a means to 'select' other Kconfig options based on what a toolchain supports or not.

For example, how would you do the `select TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` than?

Reply:
in such cases you can still add the following code to the toolchain specific Kconfig.

For example
```
config TOOLCHAIN_ARCMWDT
    select TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
```
without a type on the `TOOLCHAIN_ARCMWDT`.

That has the advantage that if a toolchain is renamed (or a spelling mistake), then because there is no type on this setting, then Kconfig will complain.
This way it's ensured that the toolchain name used in CMake and Kconfig are always in sync.

Altenatively, because `TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` is promptless (except in the cross-compile case) and `<toolchain>/Kconfig` is only sourced when that specific toolchain is in use, then the following is also possible:
```
config TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
      default y
```

Reply:
I can introduce:

```
config TOOLCHAIN_$(ZEPHYR_TOOLCHAIN_VARIANT_UPPER)
	def_bool y
```

to handle the checking you suggested, I think we will still end up with a Kconfig.<toolchain> anyways.

I don't want to default `TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` to y.  I think in general toolchain featuers should be `default n` and have the toolchains explicitly select them.

Reply:
yeah, the fact that disabling TLS support in gcc got cargo-culted into so many toolchains is really frustrating. It has *zero* impact on code not using TLS (at least on ARM). There shouldn't even *be* a GCC option to do this.

Reply:
> I think we will still end up with a Kconfig. anyways.

We might, but the main advantage is that we avoid settings getting out of sync on setting names created based on variables values in CMake, like the `TOOLCHAIN_<NAME>`.

>  I think in general toolchain featuers should be default n and have the toolchains explicitly select them.

If you noticed, it would be changing the default in a toolchain specific Kconfig: `<toolchain>/Kconfig`
and because it's promptless, then it's effectively behaves the same as a `select`.
The main difference between a `select` and a `default` in this particular case is that you can only `select` not `de-select` something.

Whereas, doing a conditional `default n` or `default y` on a promptless symbol is possible, offering the possibility to decide direction.

Also, changing the default can also allow toolchains to source / inherit a common setup as more toolchain features starts to appear, for example with toolchain adjusted Kconfig defaults we can do:

```
# toolchain/Kconfig.gnu_common
#
# gcc default supports those features:

config TOOLCHAIN_SUPPORTS_FEATURE_A
      default y

config TOOLCHAIN_SUPPORTS_FEATURE_B
      default n
...
config TOOLCHAIN_SUPPORTS_FEATURE_N
      default y
```

and then in a specific toolchain:
```
# <toolchain-specific>/Kconfig

# We don't support feature H
config TOOLCHAIN_SUPPORS_FEATURE_H
      default n

# Rest of features are common with gcc
rsource "toolchain/Kconfig.gnu_common"
```

This is why the feature settings generally should be promptless, except for the `cross-compile` use-case.
That way it will be possible to have some common Kconfig which can be inherited, for example one for gnu another for llvm based toolchains, and then there can be made local adjustments in a toolchain specifc Kconfig if there is a need.